### PR TITLE
Updated the openapi_v3

### PR DESCRIPTION
**Pull request by yashodgayashan**
### Changes done

* Make application collection and draft collection same.
* Return the application-id when application is made.
* Removed the patch and made two post requests for the assign ministry and update progress.
* Add an API to get reservation information.

### CODE REVIEW

File: resources/openapi_v3.yaml [hunk 1]
```diff
@@ -403,16 +457,16 @@ paths:
                 properties:
                   reason:
                     type: string
-  /application/{application-id}/fields/{field-id}/versions:
+  /application/{application-id}/progress/{progress-id}:
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:460]:
```suggestion
  /application/{application-id}/status/{status-id}:
```

File: resources/openapi_v3.yaml [hunk 2]
```diff
@@ -269,10 +321,7 @@ paths:
         content:
           application/json:
             schema:
-              anyOf:
-                - $ref: "#/components/schemas/AssignedMinistry"
-                - $ref: "#/components/schemas/Progress"
-                - $ref: "#/components/schemas/Field"
+              $ref: "#/components/schemas/Progress"
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:324]: Better have **status** instead of **progress**

File: resources/openapi_v3.yaml [hunk 3]
```diff
@@ -257,8 +257,60 @@ paths:
...
+        "500":
+          description: Server error(Internal Server Error).
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  reason:
+                    type: string
+  /applications/{application-id}/update-progress:
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:311]:
```suggestion
  /applications/{application-id}/update-status:
```

File: resources/openapi_v3.yaml [hunk 4]
```diff
@@ -257,8 +257,60 @@ paths:
...
+    post:
+      description: Update the progress of an application.
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:313]:
```suggestion
      description: Update the status of an application.
```

File: resources/openapi_v3.yaml [hunk 5]
```diff
@@ -403,16 +457,16 @@ paths:
                 properties:
                   reason:
                     type: string
-  /application/{application-id}/fields/{field-id}/versions:
+  /application/{application-id}/progress/{progress-id}:
     get:
-      description: Get all the versions of a field.
+      description: Get all the versions of a progress.
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:462]:
```suggestion
      description: Get the details of a status update to an application
```

File: resources/openapi_v3.yaml [hunk 6]
```diff
@@ -403,16 +457,16 @@ paths:
...
       parameters:
         - name: application-id
           in: path
           required: true
           schema:
             type: string
-        - name: field-id
+        - name: progress-id
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:469]:
```suggestion
        - name: status-id
```

File: resources/openapi_v3.yaml [hunk 7]
```diff
@@ -425,7 +479,7 @@ paths:
               schema:
                 type: array
                 items:
-                  $ref: "#/components/schemas/Field"
+                  $ref: "#/components/schemas/Progress"
```
Comment (sanjulamadurapperuma) [inline resources/openapi_v3.yaml:482]: Change this to status as well

File: resources/openapi_v3.yaml (hunk 2 above)
Comment (yashodgayashan) [reply resources/openapi_v3.yaml:324]: I will change those.

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma update the code with requested changes

**Pull request merged**